### PR TITLE
Networking/CTP-327-sync-item-usage-counts

### PR DESCRIPTION
**Pull request by erendgrmnc**

### CODE REVIEW

File: CSC8503/InventoryBuffSystem/PlayerInventory.cpp [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+#include "../SceneManager.h"
 #include "../CSC8503/LevelManager.h"
 
+namespace {
+	constexpr int DEFAULT_ITEM_USAGE_COUNT = 0;
+	
+}
+
+namespace NCL::CSC8503
+{
```
Comment (benblack90) [inline CSC8503/InventoryBuffSystem/PlayerInventory.cpp:16]: curly braces

Comment (erendgrmnc) [reply CSC8503/InventoryBuffSystem/PlayerInventory.cpp]: Done 👍

File: CSC8503/DebugNetworkedGame.cpp [hunk 2]
```diff
@@ -428,3 +443,8 @@ void DebugNetworkedGame::SyncPlayerList(){
 
 void DebugNetworkedGame::SetItemsLeftToZero(){
 }
+
+void DebugNetworkedGame::HandlePlayerEquippedItemChange(ClientSyncItemSlotPacket* packet) const {
+    int localPlayerID = static_cast<NetworkPlayer*>(mLocalPlayer)->GetPlayerID();
+    mLevelManager->GetInventoryBuffSystem()->GetPlayerInventoryPtr()->ChangePlayerItem(packet->playerID, localPlayerID, packet->slotId, static_cast<PlayerInventory::item>(packet->equippedItem), packet->usageCount);
```
[the remark this answers is not in the document]
Comment (erendgrmnc) [reply CSC8503/DebugNetworkedGame.cpp]: Done 👍

File: CSC8503/DebugNetworkedGame.h [hunk 3]
```diff
@@ -4,6 +4,11 @@
 #include "GameSceneManager.h"
 #include "NetworkedGame.h"
 
+namespace NCL::CSC8503
+{
```
[the remark this answers is not in the document]
Comment (erendgrmnc) [reply CSC8503/DebugNetworkedGame.h]: Done 👍

**Pull request merged**